test/modules/backup-gcs/backup_backend_test.go

```diff
@@ -63,19 +63,17 @@ func moduleLevelStoreBackupMeta(t *testing.T) {
 	metadataFilename := "backup.json"
 	gcsUseAuth := "false"
 
-	t.Run("setup env", func(t *testing.T) {
-		require.Nil(t, os.Setenv(envGCSEndpoint, endpoint))
-		require.Nil(t, os.Setenv(envGCSStorageEmulatorHost, endpoint))
-		require.Nil(t, os.Setenv(envGCSCredentials, ""))
-		require.Nil(t, os.Setenv(envGCSProjectID, projectID))
-		require.Nil(t, os.Setenv(envGCSBucket, bucketName))
-		require.Nil(t, os.Setenv(envGCSUseAuth, gcsUseAuth))
-
-		moduleshelper.CreateGCSBucket(testCtx, t, projectID, bucketName)
-	})
+	t.Log("setup env")
+	t.Setenv(envGCSEndpoint, endpoint)
+	t.Setenv(envGCSStorageEmulatorHost, endpoint)
+	t.Setenv(envGCSCredentials, "")
+	t.Setenv(envGCSProjectID, projectID)
+	t.Setenv(envGCSBucket, bucketName)
+	t.Setenv(envGCSUseAuth, gcsUseAuth)
+	moduleshelper.CreateGCSBucket(testCtx, t, projectID, bucketName)
 
 	t.Run("store backup meta in gcs", func(t *testing.T) {
-		require.Nil(t, os.Setenv("BACKUP_GCS_BUCKET", bucketName))
+		t.Setenv("BACKUP_GCS_BUCKET", bucketName)
```

Comment:
It looks like this env var is not propagating properly. You can see from the error in the last pipeline run:
```
Expected nil, but got: &googleapi.Error{Code:404, Message:"", Details:[]interface {}(nil), Body:"{\"code\":404,\"message\":\"path /storage/v1/b was not found\"}", Header:http.Header{"Content-Length":[]string{"57"}, "Content-Type":[]string{"application/json"}, "Date":[]string{"Tue, 21 Feb 2023 16:26:30 GMT"}, "Vary":[]string{"Origin"}}, Errors:[]googleapi.ErrorItem(nil)}
```

That `/storage/v1/b was not found`.  The bucket url is created with the value of `BACKUP_GCS_BUCKET`

Looking at the [golang source code](https://github.com/golang/go/blob/master/src/testing/testing.go#L1265-L1282) I believe that since this `t.Setenv` is being called in a nested block, that the env var is being cleaned up as soon as this block exits.

Reply:
:thinking:  But it is `backup-gcs/backup_journey_test.go` that's failing. And the tests there set their own `BACKUP_GCS_BUCKET`

![image](https://user-images.githubusercontent.com/20135478/220411838-a402e567-14ab-4563-a383-208fb2143161.png)

Reply:
Hmm...I think the test is flaky. You can see the job run on my fork where the test passed, and i changed nothing except the the job runner from `ubuntu-latest-8-cores` to `ubuntu-latest`.

https://github.com/Juneezee/weaviate/actions/runs/4235120115/jobs/7358460666#step:4:468

Reply:
All green now, you seemed to be unlucky with the flakiness